kapitan/targets.py

```diff
@@ -306,6 +306,43 @@ def load_target_inventory(inventory_path, targets):
     return target_objs
 
 
+def search_targets(inventory_path, targets, labels):
+    """returns a list of targets where the labels match, otherwise just return the original targets"""
+    if not labels:
+        return targets
+
+    try:
+        labels_dict = dict(label.split('=') for label in labels)
+    except ValueError:
+        logger.error("Compile error: Failed to parse labels, should be formatted like: kapitan compile -l env=prod app=example")
```

Comment:
Raise CompileError instead (example https://github.com/deepmind/kapitan/blob/master/kapitan/targets.py#L80)

Reply:
Not sure if we should raise here as it would print a stacktrace, we only want to print a simple message with a hint.

Reply:
CompileError is clean in this case since it is triggered when compilation fails and no stack trace should appear

Reply:
Gotcha, making another PR now.